### DISCUSSION TITLE
Confusion on `useScrollViewOffset` in 3.10.1

### DISCUSSION THREAD

**Discussion by thebiltheory** [software-mansion/react-native-reanimated#6624] [Q&A]
I am using `useScrollViewOffset` on a `Flatlist`, I am assuming that a Flatlist is also a scroll type component. 

However, (according to typescript) it seems that `useScrollViewOffset` is only meant to be used on a `ScrollView`?

<img width="537" alt="Screenshot 2024-10-23 at 11 07 20 AM" src="https://github.com/user-attachments/assets/c5458981-d210-4b40-8b37-078f4b165773">

<img width="508" alt="Screenshot 2024-10-23 at 11 11 04 AM" src="https://github.com/user-attachments/assets/4cbe76f8-59e8-4a2f-b6c4-9f66df83661e">

Yet, when I use it on a `Flatlist`, it works. Are the typings wrong, or the fact it also works on a `Flatlist` coincidental.

**Comment by MatiPl01** [comment]
Hey!

It is just the incorrect type. `useScrollViewOffset` works with other animated scrollables as well, so `Animated.Flatlist` works fine.

Remarks from the `scrollTo` function [documentation](https://docs.swmansion.com/react-native-reanimated/docs/scroll/scrollTo#remarks) apply to the `useScrollViewOffset`, so it should work with other animated scrollables, like Animated FlashList.

You can just cast the type for now to satisfy typescript. We will have to fix types in the library in the future and change the hook name to something more meaningful because the current one suggests that it works only with the `ScrollView` component.
